main/utils/markdown_parser.c

```diff
@@ -0,0 +1,1873 @@
+#include "markdown_parser.h"
```

Comment:
Did you write this or grab it from somewhere?

Reply:
That's mine. I did see some smallish C markdown parsers out there but I also wanted a couple extra features that aren't quite markdown (and also implemented most of this before actually looking around...)